pkg/zabbix_api.go

```diff
@@ -112,16 +112,22 @@ func (ds *ZabbixDatasource) BuildResponse(result *simplejson.Json) (*datasource.
 func (ds *ZabbixDatasource) ZabbixRequest(ctx context.Context, dsInfo *datasource.DatasourceInfo, method string, params *simplejson.Json) (*simplejson.Json, error) {
 	zabbixUrl := dsInfo.GetUrl()
 
-	// Authenticate first
-	if zabbixAuth == "" {
+	var result *simplejson.Json
+	var err error
+
+	for attempt := 0; attempt <= 3; attempt++ {
+		// Authenticate
 		auth, err := ds.loginWithDs(ctx, dsInfo)
 		if err != nil {
 			return nil, err
 		}
 		zabbixAuth = auth
+		result, err = ds.zabbixAPIRequest(ctx, zabbixUrl, method, params, zabbixAuth)
+		if !isNotAuthorized(err.Error()) {
```

Comment:
`err` is always `nil` at this point, so it causes a panic

Reply:
Updated 4f5f5ec